### ISSUE TITLE
Delete entire row function doesn't work

### ISSUE THREAD

**Issue by dourvaris** [OpenTSDB/asyncbigtable#29]
Found while trying to use the API in opentsdb to delete some timeseries. The code in question for opentsdb is: https://github.com/OpenTSDB/opentsdb/blob/master/src/core/TsdbQuery.java#L781

```
final DeleteRequest del = new DeleteRequest(tsdb.dataTable(), key);
```

The result of `System.out.println(del)`:

```
DeleteRequest(table="tsdb", key=[0, 0...], family="", qualifiers=[""], attempt=0, region=)
```

The function in asyncbigtable is:

https://github.com/OpenTSDB/asyncbigtable/blob/master/src/main/java/org/hbase/async/DeleteRequest.java#L69

```
  public DeleteRequest(final byte[] table, final byte[] key) {
    this(table, key, null, null, KeyValue.TIMESTAMP_NOW, RowLock.NO_LOCK);
  }
```

Exception from asyncbigtable

```
2017-11-14 13:02:03,372 ERROR [OpenTSDB I/O Worker #5] BigtableTable: Encountered exception when executing delete.
com.google.bigtable.repackaged.io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Error in field 'Mutation list' : Error in element #0 : Invalid id for collection columnFamilies : Length should be between [1,64], but found 0 ''
at com.google.bigtable.repackaged.io.grpc.Status.asRuntimeException(Status.java:503) ~[asyncbigtable-0.2.2-SNAPSHOT-jar-with-dependencies.jar:na]
at com.google.cloud.bigtable.grpc.BigtableDataGrpcClient.getBlockingResult(BigtableDataGrpcClient.java:396) ~[asyncbigtable-0.2.2-SNAPSHOT-jar-with-dependencies.jar:na]
at com.google.cloud.bigtable.grpc.BigtableDataGrpcClient.getBlockingUnaryResult(BigtableDataGrpcClient.java:358) ~[asyncbigtable-0.2.2-SNAPSHOT-jar-with-dependencies.jar:na]
at com.google.cloud.bigtable.grpc.BigtableDataGrpcClient.mutateRow(BigtableDataGrpcClient.java:255) ~[asyncbigtable-0.2.2-SNAPSHOT-jar-with-dependencies.jar:na]
at com.google.cloud.bigtable.hbase.BigtableTable.delete(BigtableTable.java:347) ~[asyncbigtable-0.2.2-SNAPSHOT-jar-with-dependencies.jar:na]
at org.hbase.async.HBaseClient.delete(HBaseClient.java:1228) [asyncbigtable-0.2.2-SNAPSHOT-jar-with-dependencies.jar:na]
at net.opentsdb.core.TsdbQuery$1ScannerCB.processRow(TsdbQuery.java:782) [tsdb-2.3.0.jar:b81ef90]
at net.opentsdb.core.TsdbQuery$1ScannerCB.call(TsdbQuery.java:750) [tsdb-2.3.0.jar:b81ef90]
at net.opentsdb.core.TsdbQuery$1ScannerCB.call(TsdbQuery.java:575) [tsdb-2.3.0.jar:b81ef90]
at com.stumbleupon.async.Deferred.doCall(Deferred.java:1278) [async-1.4.0.jar:na]
at com.stumbleupon.async.Deferred.addCallbacks(Deferred.java:688) [async-1.4.0.jar:na]
at com.stumbleupon.async.Deferred.addCallback(Deferred.java:724) [async-1.4.0.jar:na]
at net.opentsdb.core.TsdbQuery$1ScannerCB.scan(TsdbQuery.java:616) [tsdb-2.3.0.jar:b81ef90]
at net.opentsdb.core.TsdbQuery.findSpans(TsdbQuery.java:878) [tsdb-2.3.0.jar:b81ef90]
at net.opentsdb.core.TsdbQuery.runAsync(TsdbQuery.java:515) [tsdb-2.3.0.jar:b81ef90]
<snip>
```

Adding a family argument to the call makes the resulting request:

```
DeleteRequest(table="tsdb", key=[0, 0...], family="t", qualifiers=[""], attempt=0, region=)
```
Which does not throw the Exception however no rows are deleted as qualifiers is empty

Adding valid qualifiers to the argument deletes the cells as expected.

```
DeleteRequest(table="tsdb", key=[0, 0...], family="t", qualifiers=["\x00"], attempt=0, region=)
```

**Comment by dourvaris** [comment]
Seems like passing an empty family as documented here won't work on bigtable

```
  /** Special value for {@link #family} when deleting a whole row.  */
  static final byte[] WHOLE_ROW = new byte[0];
```
https://github.com/OpenTSDB/asyncbigtable/blob/master/src/main/java/org/hbase/async/DeleteRequest.java#L55

**Comment by sduskis** [comment]
This was fixed by #31
